thunder/dynamo/report.py

```diff
@@ -996,26 +1004,103 @@ def foo(x):
+
     reports = fx_report(fn, *args, **kwargs)
     if thunder_compile_kwargs is None:
         thunder_compile_kwargs = {}
-    split_reports = []
-    for fx_graph_report in reports.fx_graph_reports:
-        thunder_fx_graph_report = analyze_thunder_splits(fx_graph_report, **thunder_compile_kwargs)
-        print(f"\n{thunder_fx_graph_report.graph_name}: {thunder_fx_graph_report.split_reason}\n")
-        for split_report in thunder_fx_graph_report.subgraph_reports:
-            split_reports.append(split_report)
-    return split_reports
+    thunder_fxgraph_reports = []
+    for fxgraph_report in reports.fx_graph_reports:
+        thunder_fxgraph_report = analyze_thunder_splits(fxgraph_report, **thunder_compile_kwargs)
+        thunder_fxgraph_reports.append(thunder_fxgraph_report)
+    return thunder_fxgraph_reports
 
 
-def get_nvfusion_reports(split_reports: list[ThunderSplitGraphReport]):
-    nvfusion_reports = []
+def make_nvfusion_reports(split_reports: list[ThunderSplitGraphReport]):
+    """
+    Creates nvFusion reports for each Thunder-split report.
+    """
     for split_report in split_reports:
         split_report.create_fusion_reports()
-        for nvfusion_report in split_report.fusion_reports:
-            nvfusion_reports.append(nvfusion_report)
-    return nvfusion_reports
+
+
+# NOTE: Here we use TorchInductorSpecification (TorchInductor without TorchDynamo) to compare with Thunder. Reason: https://github.com/Lightning-AI/lightning-thunder/issues/1521
+def thunderfx_benchmark_report_from_splits(
+    thunder_fxgraph_reports: list[ThunderFXGraphReport],
+    folder_path: str | PathLike,
+    thunder_compile_kwargs: dict = None,
+    compare_fusion: bool = False,
+    rtol=0.5,
+    atol=0.0,
+):
+    """
+    A utility function that analyzes the runnability and performance benchmarks of each Thunder-split FX graph and nvFusion regions(optional). It prints out the performance metrics and saves the benchmark script in `folder_path` if the difference exceeds the tolerance (`rtol`, `atol` in seconds).
+    """
+    folder_path = Path(folder_path)
+    folder_path.mkdir(exist_ok=True, parents=True)
+    if thunder_compile_kwargs is None:
+        thunder_compile_kwargs = {}
+    thunderjit = ThunderCompileSpecification(**thunder_compile_kwargs)
+    torchinductor = TorchInductorSpecification()
+
+    runnable_split_reports: list[ThunderSplitGraphReport] = []
+    for thunder_fxgraph_report in thunder_fxgraph_reports:
+        print(f"\n{thunder_fxgraph_report.graph_name}: {thunder_fxgraph_report.split_reason}\n")
+        graph_folder = folder_path / thunder_fxgraph_report.graph_name
+        graph_folder.mkdir()
+        for split_report in thunder_fxgraph_report.subgraph_reports:
+            try:
+                split_report.run_repro(thunderjit)
+            except Exception as e:
+                print(f"Failed to run {split_report.graph_name} using Thunder with exception: {e}\n")
```

Comment:
In the future we might consider writing these to some kind of log.

Printing to stdout seems great for now